onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1114,6 +1121,42 @@ ONNX_MS_OPERATOR_SET_SCHEMA(GreedySearch, 1,
                                   GreedySearchShapeInference(ctx);
                                 }));
 
+ONNX_MS_OPERATOR_SET_SCHEMA(Sampling, 1,
+                            OpSchema()
+                                .SetDoc("Greedy Sampling for text generation.")
+                                .Attr("eos_token_id", "The id of the end-of-sequence token", AttributeProto::INT)
+                                .Attr("pad_token_id", "The id of the padding token", AttributeProto::INT)
+                                .Attr("decoder_start_token_id", "The id of the token that indicates decoding starts.", AttributeProto::INT, static_cast<int64_t>(-1))
+                                .Attr("no_repeat_ngram_size", "no repeat ngrams size", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Attr("temperature", "temperature for sampling", AttributeProto::FLOAT, 1.0f)
+                                .Attr("top_p", "top_p for sampling", AttributeProto::FLOAT, 0.0f)
+                                .Attr("filter_value", "filter value for top_p", AttributeProto::FLOAT, -1e20f)
+                                .Attr("min_tokens_to_keep", "min_tokens_to_keep", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Attr("presence_penalty", "presence penalty for sampling", AttributeProto::FLOAT, 0.0f)
+                                .Attr("custom", "absence penalty for sampling", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Attr("model_type", "model type: 0 for decoder only like GPT-2; 1 for encoder decoder like Bart", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Attr("encoder", "The subgraph for initialization of encoder and decoder. It will be called once before decoder subgraph.", AttributeProto::GRAPH, OPTIONAL_VALUE)
```

Comment:
need to add an attribute 'vocab_size' to leverage the padding optimization for vocab:
[https://github.com/microsoft/onnxruntime/pull/13842](https://github.com/microsoft/onnxruntime/pull/13842)
[https://github.com/microsoft/onnxruntime/pull/13686](https://github.com/microsoft/onnxruntime/pull/13686)